### PR TITLE
Update to Swift 5

### PR DESCRIPTION
**Pull request by RahulKatariya**
- Removes atypical/Result
 - Update ReactiveSwift to 6.0
 - Replace deprecations with latest api.

### CODE REVIEW

File: Sources/Task.swift [hunk 1]
```diff
@@ -76,15 +75,14 @@ extension Task: Hashable {
...
-		var result = launchPath.hashValue ^ (workingDirectoryPath?.hashValue ?? 0)
-		for argument in arguments {
-			result ^= argument.hashValue
+	public func hash(into hasher: inout Hasher) {
+		hasher.combine(launchPath)
+		hasher.combine(workingDirectoryPath)
+		arguments.forEach { hasher.combine($0) }
+		(environment ?? [:]).forEach { key, value in
+			hasher.combine(key)
+			hasher.combine(value)
```
Comment (RahulKatariya) [inline Sources/Task.swift]: I am not sure if the following can be used instead?

    hasher.combine(launchPath)
    hasher.combine(workingDirectoryPath)
    hasher.combine(arguments)
    hasher.combine(environment)

**Comment by ikesyo** [comment]
We should cut a release before merging this.

/cc @mdiep

File: Sources/Task.swift [hunk 2]
```diff
@@ -538,3 +532,11 @@ extension Task {
 		}
 	}
 }
+
+extension Result {
+	/// Returns a Result with a tuple of the receiver and `other` values if both
+	/// are `Success`es, or re-wrapping the error of the earlier `Failure`.
+	public func fanout<U>(_ other: @autoclosure () -> Result<U, Error>) -> Result<(Success, U), Error> {
```
Comment (mdiep) [inline Sources/Task.swift]: This shouldn't be `public`.

Comment (RahulKatariya) [reply Sources/Task.swift]: My bad. Removed the public modifier.

**Comment by mdiep** [comment]
There's also a conflict that needs to be fixed. ☺️

**Comment by RahulKatariya** [comment]
Resolved. Also updated Quick.

**Pull request merged**